Sources/swiftML/KNeighborsClassifier.swift

```diff
@@ -0,0 +1,192 @@
+import TensorFlow
+
+/// K neighbors classifier.
+///
+/// Classifier implementing the k-nearest neighbors vote.
+public class KNeighborsClassifier {
+
+    public var neighbors: Int
+    public var p: Int
+    public var weights: String
+    public var X: Tensor<Float>
+    public var y: Tensor<Float>
+  
+    /// Create a K neighbors classifier model.
+    ///
+    /// - Parameters
+    ///   - neighbors: Number of neighbors to use, default to 5.
+    ///   - weights: Weight function used in prediction. Possible values 'uniform' - uniform
+    ///     weighted, 'distance' - weight point by inverse of thier distance. default set to
+    ///     'distance'.
+    ///   - p: The distance metric to use for the tree, default set to 2.
+    public init(
+        neighbors: Int = 5,
+        weights: String = "distance",
+        p: Int = 2
+    ) {
+        precondition(neighbors > 1, "Neighbors must be greater than one.")
+        precondition(weights == "uniform" || weights == "distance",
+            "weight must be either 'uniform' or 'distance'.")
+        precondition(p > 1, "p must be greater than one.")
+
+        self.neighbors = neighbors
+        self.weights = weights
+        self.p = p
+        self.X = Tensor<Float>([0.0])
+        self.y = Tensor<Float>([0.0])
+    }
+  
+    /// fit Kneighbors classifier model.
+    ///
+    /// - Parameters
+    ///   - X: Training data Tensor<Float> of shape [number of samples, number of features].
+    ///   - y: Target value Tensor<Float> of shape [number of samples].
+    public func fit(X: Tensor<Float>, y: Tensor<Float>) {
+
+        precondition(X.shape[0] == y.shape[0], "X and y must have same number of samples.")
+        precondition(X.shape[0] > 0, "X must be non-empty.")
+        precondition(X.shape[1] >= 1, "X must have atleast single feature.")
+        precondition(y.shape[0] > 0, "y must be non-empty.")
+
+        self.X = X
+        self.y = y
+    }
+  
+    /// Return the minkowski distance based on value of p.
+    ///
+    /// - Parameters
+    ///   - a: Input tensor to find distance.
+    ///   - b: Input tensor to find distance.
+    /// - Returns: Minkowski distance based on value of p.
+    public func distance(a: Tensor<Float>, b: Tensor<Float>) -> Tensor<Float> {
```

Comment:
Please change this function to be a top-level function with no argument labels, defined in `Helpers.swift`:
```suggestion
public func minkowskiDistance(_ a: Tensor<Float>, _ b: Tensor<Float>) -> Tensor<Float> {
```

You can then reuse `minkowskiDistance` in multiple files and de-duplicate it, e.g. remove it from `KNeighborsRegressor`.